tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml

```diff
@@ -12,22 +12,22 @@ spec:
       targetPort: {{ .Values.service.port1 }}
       protocol: TCP
       name: {{ .Values.service.name1 }}
-      {{- if eq .Values.service.type "NodePort" }}
+      {{- if and .Values.service.nodePort_static (eq .Values.service.type "NodePort") }}
       nodePort: {{ .Values.global.nodePortPrefixExt | default .Values.nodePortPrefixExt }}{{ .Values.service.nodePort1 }}
       {{- end }}
     - port: {{ .Values.service.port2 }}
       targetPort: {{ .Values.service.port2 }}
       protocol: TCP
       name: {{ .Values.service.name2 }}
-      {{- if eq .Values.service.type "NodePort" }}
+      {{- if and .Values.service.nodePort_static (eq .Values.service.type "NodePort") }}
       nodePort: {{ .Values.global.nodePortPrefixExt | default .Values.nodePortPrefixExt }}{{ .Values.service.nodePort2 }}
       {{- end }}
     {{- if .Values.debug.enabled }}
     - port: {{ .Values.service.port3 }}
       targetPort: {{ .Values.service.port3 }}
       protocol: TCP
       name: {{ .Values.service.name3 }}
-      {{- if eq .Values.service.type "NodePort" }}
+      {{- if and .Values.service.nodePort_static (eq .Values.service.type "NodePort") }}
       nodePort: {{ .Values.global.nodePortPrefixExt | default .Values.nodePortPrefixExt }}{{ .Values.service.nodePort3 }}
       {{- end }}
     {{- end }}      
```

Comment:
What about the debugPorts (`service.dev.debugport`) in this application? Should we introduce nodePort_static variable for them as well?

Reply:
it is already there @rahulsharma2020, inside `{{- if .Values.debug.enabled }}` if I understand you correctly

Reply:
No, I was referring to the debug-ports defined [here](https://github.com/Telecominfraproject/wlan-cloud-helm/blob/master/tip-wlan/charts/common/templates/_ports.tpl#L14) and included in this file. These ports are used by anyone to open a debug session to an AP using Cloud. The debugPorts that you are referring are the one's to debug the application, those are not what I meant.

I don't understand the nodePortStatic role completely, hence not sure if it's applicable to the debugPorts that we opened in the portforwarding application.

Reply:
ah I see, well the idea behind nodePortStatic is to disable hardcoded nodePort ports as it prevents from deploying multiple cloudsdk on the same cluster (they would conflict on these ports), if nodePortStatic is set to false nodePort ports are chosen by k8s at the deployment time.
It can be left as is assuming that with more than one deployment on the same cluster only one of them would be switched to debug mode.

Reply:
I see. 
I am not sure how likely will K8S chooses any ports defined in the values.yaml [here](https://github.com/Telecominfraproject/wlan-cloud-helm/blob/master/tip-wlan/charts/wlan-port-forwarding-gateway-service/values.yaml#L92). They could conflict.

Also, how are you managing opening the firewall (Security Groups for Master/Worker nodes) for opening these ports (and other node ports) to the outside world?

Reply:
> I am not sure how likely will K8S chooses any ports defined in the values.yaml here. They could conflict.
if it is not a hardcoded value as it is now, it picks any free port from 30000-32767 range.

alb-ingress controller takes care of adding rule to the worker nodes security group allowing all traffic from LB security group.